### PR TITLE
examples: TTI 1st order operators

### PR DESCRIPTION
**Pull request by ofmla**
Currently, only 1st order Forward TTI operator is supported. This PR adds the missing 1st order ops (i.e., adjoint) to the AnisotropicWaveSolver class

### CODE REVIEW

**Comment by ofmla** [comment]
I committed a number of errors in this PR. Can someone please add a WIP tag, thx

**Comment by FabioLuporini** [comment]
@ofmla added

**Comment by ofmla** [comment]
I am facing some problems with the TTI 1st order Born operator and it can take a while to go to its roots and fix them. So, consider this PR as the addition of the 1st order adjoint operator.  @mloubout take a look when you get a chance

File: tests/test_gradient.py [hunk 1]
```diff
@@ -231,7 +231,7 @@ def initializer(data):
     @pytest.mark.parametrize('kernel', ['OT2'])
     @pytest.mark.parametrize('shape', [(70, 80)])
     def test_gradientJ(self, dtype, shape, kernel, space_order):
-        """
+        r"""
```
Comment (georgebisbas) [inline tests/test_gradient.py:234]: leftover ?

**Comment by FabioLuporini** [comment]
Hey @ofmla , just to be sure -- this is still unresolved right?

> I am facing some problems with the TTI 1st order Born operator and it can take a while to go to its roots and fix them

**Comment by ofmla** [comment]
> Hey @ofmla , just to be sure -- this is still unresolved right?
> 
> > I am facing some problems with the TTI 1st order Born operator and it can take a while to go to its roots and fix them

Sorry, I didn't see your message before. Yes, it is not yet solved, but the PR can be revised as it is now (only the adjoint of the 1st order forward operator). The solution to the Born issue will come in another PR I hope.

File: examples/seismic/tti/operators.py [hunk 2]
```diff
@@ -495,29 +563,36 @@ def AdjointOperator(model, geometry, space_order=4,
...
+        stagg_p = stagg_r = NODE
+    else:
+        stagg_p = stagg_r = None
 
     # Create symbols for forward wavefield, source and receivers
-    p = TimeFunction(name='p', grid=model.grid, save=None, time_order=time_order,
-                     space_order=space_order)
-    r = TimeFunction(name='r', grid=model.grid, save=None, time_order=time_order,
-                     space_order=space_order)
+    p = TimeFunction(name='p', grid=model.grid, staggered=stagg_p, save=None,
```
Comment (mloubout) [inline examples/seismic/tti/operators.py:579]: `save=None` is default can be removed.

**Review by mloubout — COMMENTED** [review]
I think it looks good overall, did you check everything still looks good? I would add the stability test for both kernels if possible.

File: tests/test_gradient.py (hunk 1 above)
Comment (georgebisbas) [reply tests/test_gradient.py:234]: oops sorry!

**Comment by ofmla** [comment]
that stability test already exists in https://github.com/devitocodes/devito/blob/master/examples/seismic/tti/tti_example.py#L44 or are you talking about another different test? I will review it again before  rebasing

**Review by mloubout — APPROVED** [review]

**Pull request merged**